Cargo.toml

```diff
@@ -11,3 +11,6 @@ strum = "0.16.0"
 strum_macros = "0.16.0"
 walkdir = "2.2.9"
 rust-embed = "5.2.0"
+codespan = "0.8.0"
+codespan-reporting = "0.8.0"
+unindent = "0.1"
```

Comment:
`codespan` provides awesome error reporting. Of course, the current implementation didn't use the full power of it, but that's not the point of this PR. `unindent` used in the experiment but seems outdated, do request changes to remove it.